packages/core/src/external/carequality/dq/process-inbound-dq.ts

```diff
@@ -26,3 +63,60 @@ export async function processInboundDocumentQuery(
     }
   }
 }
+
+async function createAndUploadCcdAndMetadata(cxId: string, patientId: string, endpointUrl: string) {
+  const { log } = out(`Generate CCD cxId: ${cxId}, patientId: ${patientId}`);
+  const queryParams = {
+    cxId,
+    patientId,
+  };
+  const params = new URLSearchParams(queryParams).toString();
+  const url = `${endpointUrl}?${params}`;
+
+  const docId = uuidv7();
+  const fileName = createUploadFilePath(cxId, patientId, `${docId}.xml`);
+
+  try {
+    const resp = await api.get(url);
+    const ccd = resp.data as string;
+    const ccdSize = sizeInBytes(ccd);
+    await s3Utils.uploadFile(bucket, fileName, Buffer.from(ccd));
+    log(`CCD uploaded into ${bucket} under this name: ${fileName}`);
+
+    const docRef: DocumentReference = {
+      resourceType: "DocumentReference",
+      status: "current",
+      subject: {
+        reference: `Patient/${patientId}`,
+      },
+      type: {
+        coding: [
+          {
+            code: "34133-9",
+            display: "SUMMARIZATION OF EPISODE NOTE",
+            system: "http://loinc.org",
+          },
+        ],
+      },
+      description: "Continuity of Care Document (C-CDA)",
+    };
+
+    const metadataFileName = createUploadMetadataFilePath(cxId, patientId, docId);
+    await createAndUploadMetadataFile({
+      s3Utils,
+      cxId,
+      patientId,
+      docId: fileName,
+      size: ccdSize.toString(),
+      docRef,
+      metadataFileName,
+      destinationBucket: bucket,
+      mimeType: "xml",
+    });
+  } catch (error) {
+    const msg = `Error creating and uploading CCD for ${cxId}/${patientId}`;
```

Comment:
Let's keep messages sent to Sentry as literals, no dynamic values; it helps Sentry to group them together - [reference](https://www.notion.so/metriport/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?pvs=4#291429931daf45a18db66d86df845749)

Reply:
omg, i keep getting burnt with this due to co-pilot 😬